### PR TITLE
Add TPC-H into CI perf testing

### PR DESCRIPTION
**Pull request by oleg-piskunov**
- creating docker image with patched Tarantool for TPC-H testing
- add TPC-H perf testing

### CODE REVIEW

File: benchs/tpch/run.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+set -o pipefail
+
+TAR_VER=$(tarantool -v | grep -e "Tarantool" |  grep -oP '\s\K\S*')
+numaopts="numactl --membind=1 --cpunodebind=1 --physcpubind=6,7,8,9,10,11"
+base_dir=$(pwd)
+
+cd /opt/tpch
+make TPC-H.db
+
+killall tarantool tpcc_load 2>/dev/null || true
```
Comment (tsafin) [inline benchs/tpch/run.sh:12]: `tpcc_load`? it's not very relevant to this script. Was it intentional, or just copy-paste?

File: benchs/tpch/run.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+sync && echo "sync passed" || echo "sync failed with error" $?
+echo 3 > /proc/sys/vm/drop_caches
+
+$numaopts /opt/tpch/bench_queries.sh 2>&1 | tee bench-sqlite.log
+make 00000000000000000000.snap
+
+killall tarantool tpcc_load 2>/dev/null || true
```
Comment (tsafin) [inline benchs/tpch/run.sh:19]: The same here - `tpcc_load` appears to be irrelevant. If not intentional.

File: benchs/tpch/run.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+base_dir=$(pwd)
+
+cd /opt/tpch
+make TPC-H.db
+
+killall tarantool tpcc_load 2>/dev/null || true
+sync && echo "sync passed" || echo "sync failed with error" $?
+echo 3 > /proc/sys/vm/drop_caches
+
+$numaopts /opt/tpch/bench_queries.sh 2>&1 | tee bench-sqlite.log
```
Comment (tsafin) [inline benchs/tpch/run.sh:16]: This is doing almost exactly the same as if you call `make bench-sqlite` but with numa options overriden. I'm ok to start with this current approach in this patch, but we should rather consider a way to fine-tune `Makefile` - to keep all internals there.
(One fine day we may change script name or add options or whatever, and this direct all would be broken since then. But no iff you would call make target)

File: benchs/tpch/run.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+make 00000000000000000000.snap
+
+killall tarantool tpcc_load 2>/dev/null || true
+sync && echo "sync passed" || echo "sync failed with error" $?
+echo 3 > /proc/sys/vm/drop_caches
+
+$numaopts tarantool execute_query.lua -n 3 2>&1 | tee bench-tnt.log
```
Comment (tsafin) [inline benchs/tpch/run.sh:23]: The same complain applies here - we should eventually modify Makefile to have nume controls embedded there.

File: benchs/tpch/run.sh (hunk 1 above)
Comment (oleg-piskunov) [reply benchs/tpch/run.sh:12]: it was copy-paste. removing.

File: benchs/tpch/run.sh (hunk 2 above)
Comment (oleg-piskunov) [reply benchs/tpch/run.sh:19]: it was copy-paste. removing

File: benchs/tpch/run.sh (hunk 3 above)
Comment (oleg-piskunov) [reply benchs/tpch/run.sh:16]: I have done changes into tpch Makefile to keep all internals there

File: benchs/tpch/run.sh (hunk 4 above)
Comment (oleg-piskunov) [reply benchs/tpch/run.sh:23]: Sure, did changes in tpch Makefile

**Comment by oleg-piskunov** [comment]
Timur, thanks a lot for review. I have made all changes you suggested.

File: benchs/tpch/run.sh [hunk 5]
```diff
@@ -7,20 +7,23 @@ numaopts="numactl --membind=1 --cpunodebind=1 --physcpubind=6,7,8,9,10,11"
...
-killall tarantool tpcc_load 2>/dev/null || true
+make create_SQL_db
+
+killall tarantool 2>/dev/null || true
 sync && echo "sync passed" || echo "sync failed with error" $?
 echo 3 > /proc/sys/vm/drop_caches
 
-$numaopts /opt/tpch/bench_queries.sh 2>&1 | tee bench-sqlite.log
-make 00000000000000000000.snap
+make bench-sqlite NUMAOPTS=$numaopts
```
Comment (tsafin) [inline benchs/tpch/run.sh:17]: The different order
```suggestion
NUMAOPTS=$numaopts make bench-sqlite
```

File: benchs/tpch/run.sh [hunk 6]
```diff
@@ -7,20 +7,23 @@ numaopts="numactl --membind=1 --cpunodebind=1 --physcpubind=6,7,8,9,10,11"
...
+
+make create_TNT_db
 
-killall tarantool tpcc_load 2>/dev/null || true
+killall tarantool 2>/dev/null || true
 sync && echo "sync passed" || echo "sync failed with error" $?
 echo 3 > /proc/sys/vm/drop_caches
 
-$numaopts tarantool execute_query.lua -n 3 2>&1 | tee bench-tnt.log
+make bench-tnt NUMAOPTS=$numaopts
```
Comment (tsafin) [inline benchs/tpch/run.sh:25]:
```suggestion
NUMAOPTS=$numaopts make bench-tnt
```

File: benchs/tpch/run.sh (hunk 5 above)
Comment (tsafin) [reply benchs/tpch/run.sh:17]: Never mind - the original version will work either

File: benchs/tpch/run.sh (hunk 6 above)
Comment (tsafin) [reply benchs/tpch/run.sh:25]: Yup, wil work either way

File: dockerfiles/ubuntu_tnt_tpch [hunk 7]
```diff
@@ -14,4 +14,4 @@ RUN ( cmake . -DENABLE_DIST=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo >build.log 2>&1
     ( cat build.log && false )
 
 # benchmarks runners
-RUN git clone https://github.com/tarantool/bench-run.git --branch=opiskunov/add-tpch /opt/bench-run
+RUN git clone https://github.com/tarantool/bench-run.git /opt/bench-run
```
Comment (avtikhon) [inline dockerfiles/ubuntu_tnt_tpch:17]: Sqaush commit with a0c076b751a74caf317efb0690bd5d059812fb67

File: dockerfiles/ubuntu_tnt [hunk 8]
```diff
@@ -9,7 +9,7 @@ RUN ( cmake . -DENABLE_DIST=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo >build.log 2>&1
     ( cat build.log && false )
 
 # benchmarks runners
-RUN git clone https://github.com/tarantool/bench-run.git --branch=opiskunov/add-tpch /opt/bench-run
+RUN git clone https://github.com/tarantool/bench-run.git /opt/bench-run
```
Comment (avtikhon) [inline dockerfiles/ubuntu_tnt]: Squash this commit with a0c076b751a74caf317efb0690bd5d059812fb67

File: dockerfiles/ubuntu_tnt_tpch [hunk 9]
```diff
@@ -5,8 +5,8 @@ COPY . /opt/tarantool
 WORKDIR /opt/tarantool
 
 # patching Tarantool sources for using with TPC-H
-patch -p1 < /opt/tpch/patches/v1-0001-Restore-partial-date-time-support-in-Tarantool-SQ.patch
-patch -p1 < /opt/tpch/patches/v1-0002-Restore-date-datetime-tests.patch
+RUN patch -p1 < /opt/tpch/patches/v1-0001-Restore-partial-date-time-support-in-Tarantool-SQ.patch
+RUN patch -p1 < /opt/tpch/patches/v1-0002-Restore-date-datetime-tests.patch
```
Comment (avtikhon) [inline dockerfiles/ubuntu_tnt_tpch:9]: Squash with commit 209968e856fea728340f991507a27fc6e2e13f2a
